src/character.cpp

```diff
@@ -5683,31 +5683,32 @@ void Character::try_reduce_weariness( const float exertion )
     weary.tick_counter++;
     if( exertion == NO_EXERCISE ) {
         weary.low_activity_ticks++;
-        // Recover twice as fast at rest
+        // Recover twice as fast if asleep/similar
         if( in_sleep_state() ) {
             weary.low_activity_ticks++;
         }
     }
 
     const float recovery_mult = get_option<float>( "WEARY_RECOVERY_MULT" );
 
-    if( weary.low_activity_ticks >= 6 ) {
+    if( weary.low_activity_ticks >= 1 ) {
         int reduction = weary.tracker;
         const int bmr = base_bmr();
-        // 1/20 of whichever's bigger
+        // 1/120 of whichever's bigger
         if( bmr > reduction ) {
-            reduction = bmr * recovery_mult;
+            reduction = std::floor( bmr * recovery_mult * weary.low_activity_ticks / 6 );
         } else {
-            reduction *= recovery_mult;
+            reduction = std::ceil( reduction * recovery_mult * weary.low_activity_ticks / 6 );
         }
         weary.low_activity_ticks = 0;
```

Comment:
This makes sleeping as effective as any NO_EXERCISE activity.
Perhaps this should be a while loop, decrementing low_activity_ticks by 1 each time?
(I'm a little suspect of the logic of unconditionally reducing ticks to 0 when >= 6, but it'd only ever be over by one previously).

Reply:
Actually, since it doubles the low activity ticks, it will double the reduction amount (barring roundoff). I was thinking about a while loop myself, but I realized that it would be equivalent to do the multiplication by the `weary.low_activity_ticks` - and will also make it more possible to substitute a float (the original weariness proposal had some `LIGHT_EXERCISE` cases giving some benefit).

Reply:
Ah, I didn't even think about that. Good point.

Reply:
I've been staring at `try_reduce_weariness()` quite a bit... plus looking at the original proposal to try to figure out the logic behind parts of it.